### ISSUE TITLE
Move job to run separately per organisation

### ISSUE THREAD

**Issue by vinayvenu** [avniproject/avni-etl#48]
### Acceptance criteria
- ETL job history for each organisation/group needs to be available so that this can be shared on a per-organisation basis
- Jobs per organisation should still be run only once an hour
- Failure logs should be available and queryable separately
- Jobs should not block each other or cause deadlocks in the system (ensure jobs do not run concurrently etc)
- When has_analytics_db is turned on/off for an org, then ETL is scheduled/unscheduled for that organisation
- If ETL service is down during this time, then has_analytics_db switch cannot be turned on or off
- Org creation/edit should work with has_analytics_db status "Unknown" if ETL service is not running. Rest of the functionality should work fine

### Tech notes
- Use API call to ETL service when has_analytics_db is toggled
- Do not use the REST API. Use regular API and wrap with our call (with admin privileges) 
- Remove has_analytics_db flag.  Use/create API to figure out if ETL is turned on/scheduled for an org (with admin privs)
- Verify if 2 orgs can run in parallel

### Out of scope
- Public availability of job run history

**Comment by petmongrels** [comment]
Quartz tables (https://stackoverflow.com/questions/48557255/what-does-each-table-for-quartz-scheduler-signify)

The fired_triggers provide information about the job. Additional info about run status of the job, we will have to capture ourselves.

We can have a manual step that creates a recurring job (simple trigger) for an organisation (quartz allows carrying information in the job detail). We can setup once every hour. Along with that we can do resource allocation by restricting the number of threads available to the scheduler to trigger jobs - http://www.quartz-scheduler.org/documentation/quartz-2.3.0/configuration/ConfigThreadPool.html
This job information can be picked up from quartz and displayed to the user.

We will have to test our how many concurrent jobs (for different organisation) we can run. Higher this number more frequently we can run the job per organisation.

If we want the user to be able start a job for an organisation immediately, we can do this instead of using locks (http://www.quartz-scheduler.org/documentation/quartz-2.3.0/cookbook/ScheduleStoredJob.html)

**Comment by petmongrels** [comment]
There are a few libraries available at the JPA, REST, and application level. There may not be that much value in using them though.

**Comment by petmongrels** [comment]
commits in web-app, etl and avni-server

**Comment by vinayvenu** [comment]
Added a documentation card on readme - #58

**Comment by vinayvenu** [comment]
I am not sure if the history queries are working fine. The response is a date, not a datetime. It also gives exactly one date, which is wrong because we have a "limit 1" in the query. 

Naming confusion - entityUuid, id are both used interchangeably in the controllers. Also, it is hard to identify that the id mentioned here is really the organisation id (hard to navigate this through the API)

The name "entityUuid" in the responses make it hard to guess that this is the identity of the org/org group. 

deleteAllJobs() is dangerous since there is no way to recover from an accidental use. Suggest removing completely. 

EtlJobHistoryItem can contain whether job was a success or not

error_message does not get populated in scheduled_job_run right now. How do exceptions get passed into the Listener?

Not sure how db connection management happens in the job when parts of the job use Jdbc and some use Hibernate

**Comment by petmongrels** [comment]
1. fixed
2. from job data perspective there can be two types of entities and one type of job. In our db there can other type of jobs as well. Hence for when one looks from db it all three info is required. In EtlJobController the job type is always ETL - renamed the id to entity uuid.
3. the uuid will be for the requested job entity type. there was no need for entity type in the response as it is used from the web app.
4. removed will do it via sql
5. done

**Comment by himeshr** [comment]
Made code changes to : 
1. Format the start and end date fields in ETLJobHistory to also include the time information
**New Output format:-**
```
    {
        "startedAt": "2023-06-15T18:30:00.000",
        "endedAt": "2023-06-25T18:30:00.000",
        "errorMessage": null,
        "success": true
    },
```
2. Set the ScheduledJobRun.startedAt value based on FireTime accessor methods rather than scheduledTime
